template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Areas/IdentityService/CertificateLoader.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+
+namespace Company.WebApplication1.Identity
+{
+    public class CertificateLoader
+    {
+        private const string File = "File";
+        private const string Store = "Store";
+
+        public static IEnumerable<X509Certificate2> LoadCertificates(IConfiguration certificates)
+        {
+            return LoadCertificates().ToList();
+            IEnumerable<X509Certificate2> LoadCertificates()
+            {
+                var certificateSources = certificates.GetChildren();
+                foreach (var certificateSource in certificateSources)
+                {
+                    var sourceKind = certificateSource.GetValue<string>("Source");
+                    switch (sourceKind)
+                    {
+                        case File:
+                            var fileSource = new CertificateFileSource();
+                            certificateSource.Bind(fileSource);
+                            yield return LoadFromFile(fileSource);
+                            break;
+                        case Store:
+                            var storeSource = new CertificateStoreSource();
+                            certificateSource.Bind(storeSource);
+                            var certificatesFromStore = LoadFromStore(storeSource);
+                            foreach (var loadedCertificate in certificatesFromStore)
+                            {
+                                yield return loadedCertificate;
+                            }
+                            break;
+                    }
+                }
+
+                X509Certificate2 LoadFromFile(CertificateFileSource source)
+                {
+                    var certificate = TryLoad(X509KeyStorageFlags.DefaultKeySet, out var error) ??
+                        TryLoad(X509KeyStorageFlags.UserKeySet, out error) ??
+                        TryLoad(X509KeyStorageFlags.EphemeralKeySet, out error);
+
+                    if (error != null)
+                    {
+                        throw error;
+                    }
+
+                    return certificate;
+
+                    X509Certificate2 TryLoad(X509KeyStorageFlags flags, out Exception exception)
```

Comment:
This whole class is going away and will be replaced with a framework component once https://github.com/aspnet/MetaPackages/issues/69 is fixed.